### PR TITLE
Switch teams redesign

### PR DESCRIPTION
**Pull request by FidelisClayton**
teams page redesign
teams not found

### CODE REVIEW

**Comment by FidelisClayton** [comment]
I kept both ways to access the teams to show that the test succeeds with the old way (using form and select input) but with the cards (that use form too) don't pass on tests.

File: app/views/teams/index.html.erb [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
+    <h4 class="page-header-title darkgrey-2">
+      <i class="mi md-20">dashboard</i> <%= t('teams.select') %>
+      <%= link_to t('teams.create'), new_team_path, class: "btn btn-link pull-right unstyled-link" %>
+    </h4>
+  </div>
+
+  <% @teams.each_with_index do |team, index| %>
+    <%= link_to switch_teams_path(team_slug: team.slug), class: 'card-link' do %>
+      <div class="col-md-4 col-xs-6">
+        <div class="panel panel-default card clickable scaleIn-animation" style="animation-delay: <%= index.to_f * 0.15 %>s" data-form-submit>
```
Comment (daniloisr) [inline app/views/teams/index.html.erb]: quick note:
you can replace `index.to_f * 0.15` with just `index* 0.15` and the result will be a float

Comment (FidelisClayton) [inline app/views/teams/index.html.erb]: Thanks!

File: app/assets/stylesheets/_card.scss [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    font-weight: bold;
+    color: $darkgrey-2;
+    border-bottom: 1px solid $lightgrey-11;
+
+    .mi {
+      z-index: 1;
+      position: relative;
+      display: inline;
+      cursor: pointer;
+      padding: 0px 5px;
```
Comment (PedroFelipe) [inline app/assets/stylesheets/_card.scss]: `0px` could be `0`

File: app/assets/stylesheets/_screen.scss [hunk 3]
```diff
@@ -623,3 +624,42 @@ div#keycut-help {
     margin-bottom: 5px;
   }
 }
+
+.project-stories {
```
Comment (PedroFelipe) [inline app/assets/stylesheets/_screen.scss]: I think that move it to a `_project-stories.scss` file would be good. Things on this file are too messed up and we should actually take components out of here to their own file.

File: app/assets/stylesheets/_screen.scss [hunk 4]
```diff
@@ -623,3 +624,42 @@ div#keycut-help {
...
+  height: 100%;
+  display: inline-block;
+  position: absolute;
+  padding-top: 2px;
+}
+
+.page-header {
+  padding-bottom: 0px;
```
Comment (PedroFelipe) [inline app/assets/stylesheets/_screen.scss]: `0`

**Comment by talyssonoc** [comment]
@daniloisr gonna check it!

File: app/views/teams/index.html.erb [hunk 5]
```diff
@@ -1,21 +1,37 @@
...
-<div class="row">
-  <div class="col-xs-12 col-sm-6">
-    <div class="form-wrapper">
-      <%= form_tag(switch_teams_path, method: :get) do %>
-        <div class="field-wrapper">
-          <%= label_tag :team_slug, 'Teams', class: 'col-sm-2 control-label' %>
-          <div class="col-sm-10">
-            <%= select_tag :team_slug, options_from_collection_for_select(@teams, :slug, :name), disabled: @teams.blank? %>
+
+<div class="col-md-8 col-md-offset-2">
```
Comment (PedroFelipe) [inline app/views/teams/index.html.erb]: Will it work fine on `xs` and `sm` screens? Teams cards should be half or full width `col-xs-12` on mobile 👍

File: app/assets/stylesheets/_animations.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
+.scaleIn-animation {
```
Comment (talyssonoc) [inline app/assets/stylesheets/_animations.scss]: We should not have camelCase on CSS class names, what do you think about simply `scale-in`? Adding `animation` to the class name also doesn't add semantics.

File: app/assets/stylesheets/_card.scss [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        content: ' ';
+        position: absolute;
+        width: 26px;
+        height: 26px;
+        left: 2px;
+        top: -3px;
+        border-radius: 30px;
+        background-color: $darkgrey-4;
+        z-index: -1;
+        transition: 0.2s cubic-bezier(0.4, 0, 0.2, 1) all;
```
Comment (PedroFelipe) [inline app/assets/stylesheets/_card.scss]: Same question as above to include Autoprefixer.

File: app/assets/stylesheets/_card.scss [hunk 8]
```diff
@@ -0,0 +1,89 @@
+.card {
+  border-radius: 0;
+  box-shadow: 0 1px 1px rgba(0,0,0,0.1);
```
Comment (talyssonoc) [inline app/assets/stylesheets/_card.scss]: This color should be on a variable, SASS allows you to do `rgba($some-color, 0.1)` 😄

File: app/assets/stylesheets/_animations.scss [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  transform: scale(0);
+  animation-name: scaleIn;
+  animation-duration: 0.3s;
+  animation-fill-mode: forwards;
+  animation-iteration-count: 1;
+}
+
+@keyframes scaleIn{
```
Comment (talyssonoc) [inline app/assets/stylesheets/_animations.scss]: Since you're using a cubic for the panel heading you could use it here too instead of using an animation. I think what you're trying to do can be achievable with the [easeOutBack](http://easings.net/#easeOutBack).

File: app/assets/stylesheets/_card.scss [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    cursor: pointer;
+  }
+
+  .panel-heading {
+    background-color: inherit;
+    font-weight: bold;
+    color: $darkgrey-2;
+    border-bottom: 1px solid $lightgrey-11;
+
+    .mi {
```
Comment (talyssonoc) [inline app/assets/stylesheets/_card.scss]: We should not add behavior to classes coming from a library, if eventually the library change the name of the class and we change it on the view it'll stop working. Add a new class for that 😄

File: app/assets/stylesheets/_screen.scss [hunk 11]
```diff
@@ -623,3 +624,42 @@ div#keycut-help {
...
+
+  .page-header-title {
+    font-weight: 600;
+  }
+}
+
+.simple-alert {
+  .mi {
```
Comment (talyssonoc) [inline app/assets/stylesheets/_screen.scss]: Same thing here about adding new behavior to library classes.

File: app/assets/stylesheets/_screen.scss [hunk 12]
```diff
@@ -623,3 +624,42 @@ div#keycut-help {
...
+.unstyled-link {
+  text-decoration: none;
+
+  &:hover {
+    text-decoration: none;
+  }
+}
+
+.darkgrey-2 {
+  color: $darkgrey-2;
```
Comment (talyssonoc) [inline app/assets/stylesheets/_screen.scss]: Why is that needed? Why not add the color to the class where you expect it to have this color?

File: app/assets/stylesheets/_screen.scss [hunk 13]
```diff
@@ -623,3 +624,42 @@ div#keycut-help {
...
+}
+
+.uppercase {
```
Comment (talyssonoc) [inline app/assets/stylesheets/_screen.scss]: Samething here about the `.darkgrey-2` color.

File: app/assets/stylesheets/application.scss [hunk 14]
```diff
@@ -20,3 +20,5 @@
 @import "project-stories";
 @import "navbar";
 @import "flash-message";
+@import 'animations';
```
Comment (talyssonoc) [inline app/assets/stylesheets/application.scss]: Let's use double quotes to keep the standard of the rest of the file.

File: app/views/teams/index.html.erb (hunk 5 above)
Comment (talyssonoc) [inline app/views/teams/index.html.erb]: Shouldn't this column be inside of a row? It may break because of the negative margin Bootstrap adds to cols.

File: app/views/teams/index.html.erb [hunk 15]
```diff
@@ -1,21 +1,37 @@
...
+  <div class="page-header">
+    <h4 class="page-header-title darkgrey-2">
+      <i class="mi md-20">dashboard</i> <%= t('teams.select') %>
```
Comment (talyssonoc) [inline app/views/teams/index.html.erb]: `dashboard` should be on locale files too.

File: app/views/teams/index.html.erb [hunk 16]
```diff
@@ -1,21 +1,37 @@
...
+      <%= link_to t('teams.create'), new_team_path, class: "btn btn-link pull-right unstyled-link" %>
+    </h4>
+  </div>
+
+  <% @teams.each_with_index do |team, index| %>
+    <%= link_to switch_teams_path(team_slug: team.slug), class: 'card-link' do %>
+      <div class="col-md-4 col-xs-6">
+        <div class="panel panel-default card clickable scaleIn-animation" style="animation-delay: <%= index* 0.15 %>s" data-form-submit>
```
Comment (talyssonoc) [inline app/views/teams/index.html.erb]: Why is this `style` prop needed?

File: config/locales/en.yml [hunk 17]
```diff
@@ -269,6 +269,7 @@ en:
     new_instruction: |
       <p>If you don't have a registered account, you can sign up first.</p>
       <p>The first user to sign in to this new team will become the administrator.</p>
+    no_teams_found: "Oops! You don't have teams."
```
Comment (talyssonoc) [inline config/locales/en.yml]: I think "You don't have teams" seems to informal/casual, what do you think about "Oops! You're not enrolled to a team"?

File: spec/support/integration_helpers.rb [hunk 18]
```diff
@@ -5,7 +5,7 @@ def sign_in(user, password = 'password')
     fill_in "Email",    with: user.email
     fill_in "Password", with: password
     click_button 'Sign in'
-    click_button 'Select Team'
+    find(:css, '.card-link', match: :first).click
```
Comment (talyssonoc) [inline spec/support/integration_helpers.rb]: Integration tests should act like the user, and the user doesn't know the CSS class of a element. Don't we have a text for Capybara to click on?

File: app/views/teams/index.html.erb (hunk 15 above)
Comment (PedroFelipe) [inline app/views/teams/index.html.erb]: This is an icon and not a string, kinda like a `class="fa fa-dashboard"` from Font Awesome. Shitty, but the way that Google's framework does.

File: config/locales/en.yml (hunk 17 above)
Comment (PedroFelipe) [inline config/locales/en.yml]: "Oops! You're not enrolled to a team `yet. Click on X to create a team`."

File: spec/support/integration_helpers.rb (hunk 18 above)
Comment (PedroFelipe) [inline spec/support/integration_helpers.rb]: The entire card element is clickable 🤔

File: app/views/teams/index.html.erb (hunk 15 above)
Comment (FidelisClayton) [inline app/views/teams/index.html.erb]: `dashboard` is an icon name

Comment (talyssonoc) [inline app/views/teams/index.html.erb]: Oh 😢 , so it's ok! Pretty disappointed with Google here hahaha

File: app/views/teams/index.html.erb (hunk 16 above)
Comment (FidelisClayton) [inline app/views/teams/index.html.erb]: This is to delay the animation between each card based on `index`.

Comment (talyssonoc) [inline app/views/teams/index.html.erb]: We can have classes for that instead and keep the behavior on CSS, like `animation-delay-1` and so on, what do you think?

File: app/views/teams/index.html.erb [hunk 19]
```diff
@@ -1,19 +1,46 @@
...
+  <div class="col-md-8 col-md-offset-2">
+    <div class="page-header">
+      <h4 class="page-header-title darkgrey-2">
+        <i class="mi md-20">dashboard</i> <%= t('teams.select') %>
+        <%= link_to t('teams.create'), new_team_path, class: "btn btn-link pull-right unstyled-link" %>
+      </h4>
+    </div>
+
+    <div class="teams">
+      <% if @teams.size > 0 %>
```
Comment (talyssonoc) [inline app/views/teams/index.html.erb]: You can use `unless @teams.empty?` here 😄

File: app/assets/stylesheets/_card.scss [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+        content: ' ';
+        position: absolute;
+        width: 26px;
+        height: 26px;
+        left: 2px;
+        top: -3px;
+        border-radius: 30px;
+        background-color: $darkgrey-4;
+        z-index: -1;
+        -webkit-transition: 0.2s cubic-bezier(0.4, 0, 0.2, 1) all;
```
Comment (talyssonoc) [inline app/assets/stylesheets/_card.scss]: We should add autoprefixer.

File: app/assets/stylesheets/_card.scss [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+        transition: 0.2s cubic-bezier(0.4, 0, 0.2, 1) all;
+        -webkit-transform: scale(0);
```
Comment (talyssonoc) [inline app/assets/stylesheets/_card.scss]: With the autoprefixer you won't need the`-webkit` versions
